### PR TITLE
Add cufflinks

### PR DESCRIPTION
**Pull request by dhirschfeld**

### CODE REVIEW

**Comment by dhirschfeld** [comment]
It seems the `cufflinks` in bioconda is a totally unrelated C++ library:
https://github.com/bioconda/bioconda-recipes/blob/master/recipes/cufflinks/meta.yaml

**Comment by dhirschfeld** [comment]
***c.c.*** @santosjorge, @timkpaine - let me know if you'd like to be added as maintainers to the recipe. It will mean a little bit of spam every now and again from conda forge. So long as I'm depending on cufflinks I'm happy to keep this up to date with pypi.

File: recipes/cufflinks/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: b0413ef57b56390b8da6faf8e90172ac73772e0d4146ff80c968f22890f293a3
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
+
+requirements:
+  build:
```
Comment (marcelotrevisani) [inline recipes/cufflinks/meta.yaml]: please remove this as you are not using it

Comment (dhirschfeld) [reply recipes/cufflinks/meta.yaml]: Done!

File: recipes/cufflinks/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+{% set name = "cufflinks" %}
```
Comment (bgruening) [inline recipes/cufflinks/meta.yaml]: Can we rename this recipe, to python-cufflinks to avoid naming confusions with the bioconda package?

Comment (timkpaine) [reply recipes/cufflinks/meta.yaml]: oooof gross.

Comment (johanneskoester) [reply recipes/cufflinks/meta.yaml]: There are other examples for this in conda-forge, e.g. python-igraph, in order to distinguish it from the igraph package.

Comment (timkpaine) [reply recipes/cufflinks/meta.yaml]: I think that makes sense in places where the python wrapper has the same name I.e. kafka/kafka-python. I just think the python cufflinks is the more established package, and the other package should change names, but I suppose thats first-mover privilege.

Comment (timkpaine) [reply recipes/cufflinks/meta.yaml]: if it generates any noise or confusion in the cufflinks package, we'll just refer them over to you guys anyway, so its no inconvenience on our end.

Comment (johanneskoester) [reply recipes/cufflinks/meta.yaml]: Yeah, in order to ensure reproducibility (in the scientific sense), changing the name of the other cufflinks package is not an option.

Comment (bgruening) [reply recipes/cufflinks/meta.yaml]: > we'll just refer them over to you guys anyway, so its no inconvenience on our end.

Please rethink your attitude here, Bioconda and conda-forge are working together not against each other. Thanks.

Comment (timkpaine) [reply recipes/cufflinks/meta.yaml]: we are neither though. I'm saying if people ask "why doesn't `conda install cufflinks`" work, we will refer them to this PR.

Comment (timkpaine) [reply recipes/cufflinks/meta.yaml]: or to rephrase, "we don't have an issue with this if you don't"

Comment (bgruening) [reply recipes/cufflinks/meta.yaml]: > or to rephrase, "we don't have an issue with this if you don't"

That sounds way cooler :)
Imho, and this has historical reasons, we missed to prefix packages or invent some other namespace systems. For example, R packages we also prefix `r-` and similar for `perl-`. Now we need to deal with the consequences. Thanks for your understanding!

**Comment by dhirschfeld** [comment]
I think conda really needs proper namespacing support so that different languages don't clash. The manual prefix-namespacing is really just a  hack to work around this limitation.

This feature proposal has been around for a while but it looks like there's now some movement in that direction:
 * https://gist.github.com/mcg1969/da5aec380d2ed083b79ddcf151ca16f1
 * https://github.com/conda/conda/issues/7429
 * https://github.com/conda/conda/pull/7446

File: recipes/cufflinks/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{% set namespace = "python" %}
+{% set name = "cufflinks" %}
+{% set version = "0.14.4" %}
+
+package:
+  name: {{ namespace|lower }}-{{ name|lower }}
```
Comment (marcelotrevisani) [inline recipes/cufflinks/meta.yaml]: Please, also rename the folder to ``python-cufflinks``

**Comment by johanneskoester** [comment]
Thank you!

**Pull request merged**